Improve documentation about environment variables being used (#138)
kubernetes-sigs/kube-scheduler-simulator: docs/env-variables.md

> # Different Environment Variables used

Reviewer: Maybe we should also describe about `KUBECONFIG`?
like this.
```
`KUBECONFIG`:The cluster from which you are importing is indicated by this environment variable.
  [Read more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)
```

Author: Done with the latest push. Thanks for mentioning that.

This page describes about the different environment variables that are
used to configure kube-scheduler-simulator.

`Port`: This is the port number on which kube-scheduler-simulator is started and it's value is 1212 which is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L7)

`EtcdURL`: This is the URL for kube-scheduler-simulator etcd which runs locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L8)

`FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).

`KUBE_API_HOST`: This is the host on which the kube-apiserver serves
for the simulator server.

`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)

`KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be

> passed via this environment varia
> ble and the simulator will start the

Reviewer: ```suggestion
passed via this environment variable and the simulator will start the
```
On Github's markdown view, space was inserted into `variable`

Author: Updated.

# Different Environment Variables used

This page describes about the different environment variables that are
used to configure kube-scheduler-simulator.

`Port`: This is the port number on which kube-scheduler-simulator is started and it's value is 1212 which is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L7)

`EtcdURL`: This is the URL for kube-scheduler-simulator etcd which runs locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L8)

`FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).

`KUBECONFIG`: This variable is used to access the cluster which is

> imported locally. [Read

Reviewer: @pravarag thanks!
hmm, at least, in our simulator, this value is used to access the cluster when we'd like to import the cluster's resources. So, I think that it is not a `locally imported cluster` but a `cluster which will be imported locally`.
(But my English is not good, sorry...)
WDOT?

Author: How about, 

```
This variable is used by kubectl tool to find the information it needs to access the cluster which will be imported locally and communicate with the API server of the cluster. 
```
I took some reference from the official Kubernetes docs.

Author: Let me know your thoughts on the above

Reviewer: `KUBECONFIG`
[In general simulator]: Currently not used.
[In Beta]: Used for settings indicating clusters to be imported. To be precise, the KubeConfig file of the cluster to be imported is specified so that simulator can read the resources in the specified cluster.

So, How is this?
```
`KUBECONFIG`: Currently not used. [Beta: Existing Cluster Importing] This variable is used to find the information it needs to access the cluster which will be imported to our simulator and communicate with the API server of the cluster. If you use the Beta feature, you need to set the `Kubeconfig` file path of the cluster which will be imported. [Read more]
```

Author: Updated.

more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)

`KUBE_API_HOST`: This is the host on which the kube-apiserver serves
for the simulator server.

`KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)

`KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be
passed via this environment variable and the simulator will start the

> scheduler with that configuration.

Reviewer: Also, could you add this description?
```suggestion
scheduler with that configuration. [Beta: Existing Cluster Importing] The same is true for scheduler settings, even for beta features. This is because scheduler settings cannot be imported directly from an existing cluster.
```

Author: Done. updated as per your suggestions with the latest push.

# Different Environment Variables used

> This page describes about the different environment variables that are
> used to configure kube-scheduler-simulator.

Reviewer 2: ```suggestion
This page describes about the environment variables that are
used to configure kube-scheduler-simulator.

You can refer [docker-compose.yml](./docker-compose.yml) as an example use.
```

Author: Updated.

> `Port`: This is the port number on which kube-scheduler-simulator is started and it's value is 1212 which is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L7)

Reviewer 2: ```suggestion
`Port`: This is the port number on which kube-scheduler-simulator server is started.
```

Author: Updated.

> `EtcdURL`: This is the URL for kube-scheduler-simulator etcd which runs locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L8)

Reviewer 2: ```suggestion
`EtcdURL`: This is the URL for etcd. 
The simulator run kube-apiserver internally, and the kube-apiserver uses this etcd.
```

Author: Updated.

> `FrontendURL`: This URL is on which kube-scheduler-simulator is reachable locally and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3111ec419ccb8c49197b75385fbd166f7f159435/docker-compose.yml#L9).

Reviewer 2: ```suggestion
`FrontendURL`: This URL represents the URL web UI started.
The simulator and internal kube-apiserver set this origin as the allowed origin for CROS.
```

Reviewer 2: Sorry, it's already renamed to CORS_ALLOWED_ORIGIN_LIST. Please rename
https://github.com/kubernetes-sigs/kube-scheduler-simulator/commit/c922c36d0f5678c9f5a9c9573aa333bbed91b43e

Author: Updated.

> `KUBECONFIG`: Currently not used. [Beta: Existing Cluster Importing]
> This variable is used to find the information it needs to access the
> cluster which will be imported to our simulator and communicate with
> the API server of the cluster. If you use the Beta feature, you need
> to set the `Kubeconfig` file path of the cluster which will be
> imported. [Read more]. [Read
> more](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/#the-kubeconfig-environment-variable)

Reviewer 2: ```suggestion
`KUBECONFIG`: This is for the beta feature "Existing Cluster Importing" 
and the feature is enabled when you set the value.
Currently, the simulator tries to import ALL resources related to scheduling from your cluster,
so, please be careful (or avoid) using it in clusters with large amounts of resources. 
This variable is used to find KubeConfig it needs to access your
cluster to import resources to our simulator. 
```

Author: Made a small change in the last two lines: `Kubeconfig required to access your cluster for importing resources to scheduler simulator`. Let me know if this looks good.

> `KUBE_API_HOST`: This is the host on which the kube-apiserver serves
> for the simulator server.

Reviewer 2: ```suggestion
`KUBE_API_HOST`: This is the host of kube-apiserver which the simulator starts internally.
```

Author: Updated.

> `KUBE_API_SERVER_URL`: This is the kube-apiserver URL and it's value is set [here](https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/docker-compose.yml#L26)

Reviewer 2: This `KUBE_API_SERVER_URL` is env for frontend. And I think we should write the doc about env for frontend in another section to avoid confusion.

Please create another section (like `## For web UI`) and move `KUBE_API_SERVER_URL` in the section

Author: Done, thanks for mentioning that.

> `KUBE_SCHEDULER_CONFIG_PATH`: A KubeSchedulerConfiguration file can be
> passed via this environment variable and the simulator will start the
> scheduler with that configuration. [Beta: Existing Cluster Importing] The same is true for scheduler settings, even for beta features. This is because scheduler settings cannot be imported directly from an existing cluster.

Reviewer 2: ```suggestion
`KUBE_SCHEDULER_CONFIG_PATH`: The path to a KubeSchedulerConfiguration file. 
If passed, the simulator will start the scheduler with that configuration.
Or, if you use web UI, you can change the configuration from the web UI as well..
```

Author: Updated.

# Different Environment Variables used

This page describes about the different environment variables that are
used to configure kube-scheduler-simulator. 

Please refer [docker-compose.yml](./docker-compose.yml) as an example
use.

> `Port`: (required) This is the port number on which kube-scheduler-simulator
> server is started. It's default value is `1212`.

Reviewer 2: No default value on it.
```suggestion
`Port`: (required) This is the port number on which kube-scheduler-simulator
server is started.
```

Author: Updated.

> `EtcdURL`: (required) This is the URL for etcd. The simulator runs kube-apiserver
> internally, and the kube-apiserver uses this etcd. It's default value is `http://simulator-etcd:2379`.

Reviewer 2: No default value on it. And I guess you mean `KUBE_SCHEDULER_SIMULATOR_ETCD_URL`.

```suggestion
`KUBE_SCHEDULER_SIMULATOR_ETCD_URL`: (required) This is the URL for etcd. The simulator runs kube-apiserver
internally, and the kube-apiserver uses this etcd.
```

Author: Updated.

`FrontendURL`: This URL represents the URL web UI started. The
simulator and internal kube-apiserver set the origin as the allowed
origin for `CORS_ALLOWED_ORIGIN_LIST`.

> `KUBECONFIG`: (required) This is for the beta feature "Existing cluster

Reviewer 2: It's not required env.

Author: Updated.

Importing". This variable is used to find Kubeconfig required to
access your cluster for importing resources to scheduler simulator.

> `KUBE_API_HOST`: (required) This is the host of kube-apiserver which the

Reviewer 2: same. not required one.

Author: Updated

simulator starts internally.

> `KUBE_SCHEDULER_CONFIG_PATH`: (required) The path to a KubeSchedulerConfiguration

Reviewer 2: It's also not required.

Author: Updated.
